### PR TITLE
Add a requester pays bucket permission args to boto3 for s3 download file

### PR DESCRIPTION
**Pull request by karan6181**
## Description of changes:
- Add a requester pays bucket permission args to boto3 for s3 download file. Without the args, it would throw either 403 or 404 error.

## Merge Checklist:
_Put an `x` without space in the boxes that apply. If you are unsure about any checklist, please don't hesitate to ask. We are here to help! This is simply a reminder of what we are going to look for before merging your pull request._

### General
- [x] I have read the [contributor guidelines](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md)
- [ ] This is a documentation change or typo fix. If so, skip the rest of this checklist.
- [x] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the MosaicML team.
- [x] I have updated any necessary documentation, including [README](https://github.com/mosaicml/streaming/blob/main/README.md) and [API docs](https://github.com/mosaicml/streaming/tree/main/docs) (if appropriate).

### Tests
- [x] I ran `pre-commit` on my change. (check out the `pre-commit` section of [prerequisites](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md#prerequisites))
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate).
- [x] I ran the tests locally to make sure it pass. (check out [testing](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md#running-tests))
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes.

### CODE REVIEW

File: streaming/base/storage/download.py [hunk 1]
```diff
@@ -47,14 +48,23 @@ def _download_file(unsigned: bool = False) -> None:
     if obj.scheme != 's3':
         raise ValueError(f'Expected obj.scheme to be "s3", got {obj.scheme} for remote={remote}')
 
+    extra_args = {}
+    # When enabled, the requester instead of the bucket owner pays the cost of the request
+    # and the data download from the bucket.
+    if os.environ.get('MOSAICML_STREAMING_AWS_REQUESTER_PAYS') == '1':
```
Comment (knighton) [inline streaming/base/storage/download.py:54]: We need to specify the list of bucket names

Comment (karan6181) [reply streaming/base/storage/download.py]: Done in https://github.com/mosaicml/streaming/pull/231/commits/6a80cc6d97a762dea107a75b4c6247fff5058ced.

File: docs/source/how_to_guides/configure_cloud_storage_cred.md [hunk 2]
```diff
@@ -48,6 +48,25 @@ aws_secret_access_key=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
...
+```{code-tab} py
+import os
+os.environ['MOSAICML_STREAMING_AWS_REQUESTER_PAYS'] = 'streaming-bucket'
+
+# For more than one requester pays bucket
+os.environ['MOSAICML_STREAMING_AWS_REQUESTER_PAYS'] = 'streaming-bucket,another-bucket'
+```
+
+```{code-tab} sh
+export MOSAICML_STREAMING_AWS_REQUESTER_PAYS='streaming-bucket'
```
Comment (knighton) [inline docs/source/how_to_guides/configure_cloud_storage_cred.md:63]: don't need the single quotes

**Review by knighton — APPROVED** [review]

File: streaming/base/storage/download.py [hunk 3]
```diff
@@ -23,11 +23,12 @@ def download_from_s3(remote: str, local: str, timeout: float) -> None:
         timeout (float): How long to wait for shard to download before raising an exception.
     """
 
-    def _download_file(unsigned: bool = False) -> None:
+    def _download_file(unsigned: bool = False, extra_args: dict = {}) -> None:
```
[the remark this answers is not in the document]
Comment (karan6181) [reply streaming/base/storage/download.py]: Make sense. Done in https://github.com/mosaicml/streaming/pull/231/commits/b39e9616b3c8e4b36f59b0b96a00d9f2a497f125 .

**Pull request merged**